embeddings/data/data_loader.py

```diff
@@ -34,29 +33,30 @@ def load(self, dataset: Dataset[str]) -> datasets.DatasetDict:
 
 class HuggingFaceLocalDataLoader(DataLoader[str, datasets.DatasetDict]):
     def load(self, dataset: Dataset[str]) -> datasets.DatasetDict:
-        if isinstance(dataset, HuggingFaceDataset):
+        if isinstance(dataset, LoadableDataset):
             result = datasets.load_from_disk(str(dataset.dataset))
```

Comment:
```suggestion
            result = datasets.load_from_disk(dataset.dataset)
```
dataset is a `str` attribute